### ISSUE TITLE
[V2] WS crashes on start

### ISSUE THREAD

**Issue by LawJolla** [jimkyndemeyer/js-graphql-intellij-plugin#238]
## Version and Environment Details
__Operation system__: MacOS 10.14.4

__IDE name and version__: WS 2019.1

__Plugin version__: 2.0.0

## Actual Behaviour

Crashes WS on launch. (WS screen freeze, force restart)

## Steps to Reproduce / Link to Repo with Reproduction and Instructions

This is a mono-repo with the following config...

```
{
  "projects": {
    "admin": {
      "schemaPath": "./packages/admin-server-nexus/src/generated/schema.graphql",
      "includes": ["./packages/admin-client/src/containers/*.tsx"]
    }
  }
}
```

I assume this is a config problem, but no config I think of stops the crash.  I've downgraded in the interim.  Any help is very appreciated!

**Comment by jimkyndemeyer** [comment]
Hi Dennis.

Thanks for using the plugin. The config looks valid to me, so not sure why you're seeing this freeze.

I assume that the freeze prevents you from getting to a stack trace via the error icon in the bottom right? If so, the log file might provide the information needed to make this issue actionable. You can access the log via the "Help" menu > "Show log" menu item. Hopefully a search for graphql in that log yields an exception you can attach to this issue.

Best regards,
Jim.

**Comment by LawJolla** [comment]
Hi Jim,

[idea.log](https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/files/3062199/idea.log)
Thanks so much for your quick response!  

You're right about the stack trace.  I went through the log file, but nothing jumps out to me.  I deleted it and reran the crash, and attached that file here.  Maybe you can see something.

It appears to have a problem with the generated Nexus output.  The crash comes as soon as I point the `schemaPath` at the generated output.  Here's that file.  

https://gist.github.com/LawJolla/168b4b1e1634beed44e08ea53cc8a2a1

I wish I was bright enough to be of more help.  Thanks for your great work!

**Comment by jimkyndemeyer** [comment]
Thanks Dennis.

The schema file should help me reproduce the issue and find the underlying cause.

I'll update once I know more.

/Jim

**Comment by jimkyndemeyer** [comment]
Hi Dennis.

I have a fix prepared for the upcoming 2.0.1 release which I'm targeting for the middle of next week.

Based on my testing this was not a classic deadlock/freeze, but your schema is so deeply nested in the input type definitions that graphql-java had to build stacks of 500+ type names to untangle their dependencies and verify the schema. I've upgraded graphql-java from 11.0 to 12.0 and it appears to perform much better with your schema.

Let me know if you'd like to test this fix in separate release and I'll make one available to you. Otherwise the typical turnaround time via the JetBrains plugin repo is 1-2 business days.

Quite a schema you have by the way, it's almost double the size of the entire GitHub API.

Best regards,
Jim.

**Comment by LawJolla** [comment]
Hi Jim,

Thank you for your incredible help and work!

More broadly, my project looks massive because it incorporates 35 Prisma types and its associated CRUD fields and types.  I doubt my schema is more complicated than a run-of-the-mill Prisma application.

Did you see any problematic pattern in the generated schema that would help to modify?  I'd be happy to open an issue there and discuss it with the Prisma/Nexus team.  From my experience, few Prisma users use Jetbrains products, so I may be a bit of an oddball for the plugin.
